drizzlepac/devutils/comparison_tools/compare_sourcelists.py

```diff
@@ -1149,8 +1149,8 @@ def comparesourcelists(slNames=None, imgNames=None, good_flag_sum = 255, plotGen
         # Get platescale
         plate_scale = wcsutil.HSTWCS(imgNames[0], ext=('sci', 1)).pscale
         matched_values = extractMatchedLines("X", refData, compData, matching_lines_ref, matching_lines_img, bitmask=bitmask)
-        if output_json_filename:  # Add matched values to diag_obj
-            diag_obj.add_data_item(matched_values,"X MATCHED VALUES")
+        if output_json_filename and matched_values.shape[1] > 0:  # Add matched values to diag_obj
+            diag_obj.add_data_item(Table([matched_values[0], matched_values[1]], names=('reference', 'comparison')),"X MATCHED VALUES")
             diag_obj.add_update_info_item("header", "plate_scale", plate_scale)
     if matched_values.shape[1] > 0:
         formalTitle = "X Position"
```

Comment:
The following is a suggestion only, not a request for a change...  Having multiple copies of essentially the same code for each column being extracted comes with a couple of problems; namely, more places for errors to be made, more code for Python to interpret and more difficult to understand what is being done in the code.  It works, but it would be better to define a single function that could be called for each column ("X", "Y", ...) instead.  I could see 'extract_column(colname, formalTitle, input_json_filename, output_json_filename, regressionTestResults)' that returns the 'matchedValues' for plotting later.  I would also argue that 'colTitles' is simply 'regressionTestResults.keys()' and is not strictly necessary now that dicts are ordered, although defining 'regressionTestResults' as an OrderedDict to start with would resolve that as well.  This would let lines 1145-1164 be pulled into a single function that would then replace the same lines for all the subsequent columns.   A list of column names [('X','X Position'), ('Y', 'Y Position'),...] could then serve to call this function for all the columns, making it more obvious what is being done.  I may be overlooking some interaction between each of these sections of code, but initially this could eliminate 14 nearly identical sections of code if the unique handling of the outputs can be handled separately.  **PLEASE** consider whether this encapsulation is possible and **if not, disregard this comment!**

Reply:
There's just enough complexity and non-uniqueness between different comparisons (X position, Y position, RA/DEC positions, etc.) that encapsulating this would be complicated. Best leave it alone for right now.